Fix error caused by path value on windows

mmagonde opened the pull request:
Fails validation when path starts with '/' as expected rather than drive letter on windows

alexjfisher commented:
@mmagonde Having fun?  We never actually used this type previously?  Anyway, are you saying that you have a valid Windows path that does start with a '/'?

alexjfisher commented:
oh, and no, I don't still have commit permissions on this repo! - don't worry :)

mmagonde commented:
@alexjfisher no path is used to form part of the rest api string here https://github.com/OrdnanceSurvey/puppet-fme/blob/master/lib/puppet/provider/fme_resource/rest_client.rb#L22 so that validation should just check to see if the value starts with '/' rather than it being an absolute path. On windows the validation would only be true if the path started with the drive letter. Here means the path relative to the "shared resource".

alexjfisher commented:
@mmagonde Ah, yes, understood.  It's only `source` that refers to a local filepath.  Change looks good then I think.  Either way, I'm glad that you're using it now.  It took a couple of weeks to get working (from Linux at least!), and then we didn't need it.  To get the tests running again, you should probably just drop the puppet 3 tests (puppet 3 is EOL and you're not using it either) and move to testing puppet 4 only.  You can then unpin rspec.  I think this is where the problem is (probably).

Commit:
Drop support for puppet 3

Commit:
Fix error caused by path value on windows

Fails validation when path starts with '/' as expected rather than drive
letter on windows but the path required is relative to the "shared
resource".
